worker/changestream/manifold.go

```diff
@@ -25,9 +26,13 @@ type Logger interface {
 	IsTraceEnabled() bool
 }
 
+// MetricsCollectorFn is an alias function that allows the creation of
+// a metrics collector.
+type MetricsCollectorFn = func() *Collector
+
 // WatchableDBFn is an alias function that allows the creation of
 // EventQueueWorker.
-type WatchableDBFn = func(string, coredatabase.TxnRunner, FileNotifier, clock.Clock, Logger) (WatchableDBWorker, error)
+type WatchableDBFn = func(string, coredatabase.TxnRunner, FileNotifier, clock.Clock, *Collector, Logger) (WatchableDBWorker, error)
```

Comment:
With the changes to the proposed setup, this becomes an interface.

Reply:
yes exactly